### ISSUE TITLE
Brifit device is not recognized

### ISSUE THREAD

**Issue by blumenstiel** [lstrojny/homebridge-ble-thermobeacon#29]
**Describe Your Problem:**
I try to add a Brifit Bluetooth Thermometer Hygrometer to Homebridge but the device is not recogniced. I guess that my sensor is slightly different because the referenced scripts can read the data (my sensor https://www.amazon.de/dp/B08GWVLXZ3?ref=ppx_yo2ov_dt_b_product_details&th=1). I would be very glad for some help. Thanks!

I tested the ThermoBeacon script. The Pi is reciving the data:
```bash
raspberry@pi:~/ThermoBeacon $ python tbcmd.py scan
[3d:26:00:00:01:a6] [18] T= 20.50°C, H = 55.38%, Button:Off, Battery : 90%, UpTime =    25195s
[3d:26:00:00:01:a6] [18] T= 20.50°C, H = 55.38%, Button:Off, Battery : 90%, UpTime =    25196s
[3d:26:00:00:01:a6] [18] Max=26.50°C at 19248s, Min=12.56°C at 10838s
```
Maybe the sensor is not identified correctly as "identify" is not working:
```bash
raspberry@pi:~/ThermoBeacon $ python tbcmd.py identify -mac 3d:26:00:00:01:a6
connectd
///Characteristic 0000fff5-0000-1000-8000-00805F9B34FB was not found!
```
I tested identify with and without pressing the button. I guess this is somehow related to the Transmit Handle 0x0021 in https://github.com/rnlgreen/thermobeacon/blob/main/thermobeacon2.py

"query" is working fine:
```bash
raspberry@pi:~/ThermoBeacon $ python tbcmd.py query -mac 3d:26:00:00:01:a6
{'mac': '3d:26:00:00:01:a6', 'temp': 20.44, 'relhum': 56.06, 'button': False, 'battery': 90.32, 'uptime': 26986}
```

I also scaned for BLE with bluetoothctl and scan on and got:
```
[NEW] Device 3D:26:00:00:01:A6 ThermoBeacon
[CHG] Device 3D:26:00:00:01:A6 RSSI: -76
[CHG] Device 3D:26:00:00:01:A6 ManufacturerData Key: 0x0018
[CHG] Device 3D:26:00:00:01:A6 ManufacturerData Value:
  00 00 a6 01 00 00 26 3d a8 01 30 4b 00 00 c9 00  ......&=..0K....
  56 2a 00 00                                      V*..            
```

**Logs:**

Logs from starting the child bridge:
```bash
[03/01/2023, 22:32:09] [homebridge-ble-thermobeacon] Launched child bridge with PID 1355
[03/01/2023, 22:32:09] Registering platform 'homebridge-ble-thermobeacon.BleThermoBeacon'
[03/01/2023, 22:32:09] [homebridge-ble-thermobeacon] Loaded homebridge-ble-thermobeacon v0.0.15 child bridge successfully
[03/01/2023, 22:32:09] Loaded 0 cached accessories from cachedAccessories.0E181663E32D.
[03/01/2023, 22:32:09] Homebridge v1.6.0 (HAP v0.11.0) (homebridge-ble-thermobeacon) is running on port 47939.
```

**Plugin Config:**

```json
{
    "debug": true,
    "buttonAsLock": true,
    "devices": [
        {
            "address": "3d:26:00:00:01:a6",
            "name": "Blue",
            "buttonAsLock": true
        }
    ],
    "platform": "BleThermoBeacon",
    "_bridge": {
        "username": "0E:18:16:63:E3:2D",
        "port": 47939
    }
}
```
I tested it with and without buttonAsLock.

**Environment:**

* **Plugin Version**:
* **Homebridge Version**: v1.6.0 
* **Node.js Version**:  v18.12.1
* **NPM Version**: 8.19.2
* **Operating System**: Raspbian 
* I followed the "Configuration" from ReadMe and added bluetooth.conf.

**Comment by lstrojny** [comment]
Interesting! The name of the beacon looks right and the model is exactly the one I got as well. 
Could you run the plugin with `debug: true` in the config and attach the log as well plus run the `homebridge-blue-thermobeacon-scan` CLI tool that is automatically installed with the plugin?

**Comment by blumenstiel** [comment]
Here are the relevant logs. The sensor is recognised:
```
[04/01/2023, 10:54:12] [BleThermoBeacon] Initializing BleThermoBeacon platform...
[04/01/2023, 10:54:12] [BleThermoBeacon] Finished initializing platform: undefined
[04/01/2023, 10:54:12] [BleThermoBeacon] Executed didFinishLaunching callback
[04/01/2023, 10:54:12] Publishing bridge accessory (name: Homebridge BAEC, publishInfo: {
  username: '***',
  port: 51166,
  pincode: '***-**-***',
  category: 2,
  bind: undefined,
  mdns: undefined,
  addIdentifyingMaterial: true,
  advertiser: 'avahi'
}).
...
[04/01/2023, 10:54:35] [BleThermoBeacon] Handler BrifitThermometerHandler can handle {"uuid":"3d26000001a6","rssi":-77,"advertisement":{"localName":"ThermoBeacon","manufacturerData":{"type":"Buffer","data":[24,0,0,0,166,1,0,0,38,61,168,1,48,75,0,0,201,0,86,42,0,0]}}}
[04/01/2023, 10:54:35] [BleThermoBeacon] BrifitParser returned null for 18000000a6010000263da801304b0000c900562a0000
[04/01/2023, 10:54:35] [BleThermoBeacon] Handler BrifitThermometerHandler returned sensor data null for {"uuid":"3d26000001a6","rssi":-77,"advertisement":{"localName":"ThermoBeacon","manufacturerData":{"type":"Buffer","data":[24,0,0,0,166,1,0,0,38,61,168,1,48,75,0,0,201,0,86,42,0,0]}}}
```

I am new to js, so I don't realy know if I excecuted the tool correctly. Running with node, the tool is showing no output:
```
homebridge@pi:/var/lib/homebridge/node_modules/homebridge-ble-thermobeacon $ ls
bin  code-generation  config.schema.json  dist  jest.config.js  LICENSE  node_modules  package.json  README.md  tests
homebridge@pi:/var/lib/homebridge/node_modules/homebridge-ble-thermobeacon $ node dist/bin/homebridge-ble-thermobeacon-scan.js

```

Thanks for the help!

**Comment by lstrojny** [comment]
OK, here is what we see in the log:
 * Indeed, the sensor is found and correctly detected
 * It just happens to carry the 22 byte long message that starts with `0x18` in the manufacturer data at that moment. The 22 byte long `0x18` message encodes min/max information which is nice but not useful for our purpose (what you showed from `tbcmd.py` as `Max=26.50°C at 19248s, Min=12.56°C at 10838s`)


So, the beacon is supposed to cycle back and forth between the 20 byte message starting with `0x15` (which is the one we are interested in here) and the `0x18` message (which we need to ignore). Dumb question: if you let it run for a little bit longer (I am talking here a few minutes maybe), does it start to work?

**Comment by blumenstiel** [comment]
I let it run over night, the device is not found. When I use `tbcmd.py scan`, the message key is `18`, too. I guess that they changed something at their software. The only diffrence is the length of the `bvalue` in [tbcmd.py](https://github.com/iskalchev/ThermoBeacon/blob/f09f4719ba8ed2d6654e2656b8043ccd408d395b/tbcmd.py#L94).
```
[3d:26:00:00:01:a6] [18] T= 19.25°C, H = 59.00%, Button:Off, Battery : 90%, UpTime =   160960s
```
But at least we know why it is not working and I can fix it, thanks!
Do I just have to change [brifit.ts](https://github.com/lstrojny/homebridge-ble-thermobeacon/blob/08910bca133da2e96de80592b797be687aae89be/src/thermometer/brifit.ts#L78)?

**Comment by blumenstiel** [comment]
Removing the check for key == 15 fixed it.

**Comment by lstrojny** [comment]
Fixed in [0.0.16](https://github.com/lstrojny/homebridge-ble-thermobeacon/releases/tag/0.0.16)
